packages/i18n-utils/src/index.ts

```diff
@@ -9,11 +9,14 @@ import { useI18n } from '@automattic/react-i18n';
 import { localizeUrl } from './localize-url';
 import { Locale } from './locales';
 
-export function useI18nUtils() {
+export interface I18nUtils {
+	localizeUrl: typeof localizeUrl;
+}
+
+export function useI18nUtils(): I18nUtils {
 	const { i18nLocale } = useI18n();
 
 	return {
-		locale: i18nLocale,
 		localizeUrl: ( fullUrl: string, locale?: Locale ) => {
 			if ( locale ) {
 				return localizeUrl( fullUrl, locale );
```

Comment:
At first I didn't think we needed to support this case i.e. passing in their own value for `locale`. If they had the locale all along then maybe we should have encouraged them to use the non-hook version.

But I think this is good actually. By virtue of the fact that `useI18nUtils()` calls `useI18n()`, it means the hook will re-render when the underlying locale data changes and that makes sense.

So what I'm saying is leave it like this 😅

Reply:
I don't know why GitHub is highlighting the `locale: i18nLocale` line. I'm talking about the `return localizeUrl( fullUrl, locale );` line.

Reply:
That's a nice side effect, the reason I did this was to maintain the same api publicly as the original version, but if you pull in the function directly you should be under no assumption that the locale is optional in that context - hopefully typescript catches this as an error?

Reply:
It's highlighting because I removed the locale in a later commit, I think I might have linked you the latest version to avoid seeing all your earlier (not yet merged) changes in the PR